controllers/networkinterface_controller.go

```diff
@@ -47,36 +45,19 @@ const (
 	defaultFirewallRulePrefix = "0.0.0.0/0"
 )
 
-// workaroundNoNetworkInterfaceIPV6 is a workaround to only use ipv4 addresses.
-// remove as soon as bug is fixed.
-func workaroundNoNetworkInterfaceIPV6(ips []netip.Addr) []netip.Addr {
-	var res []netip.Addr
-	for _, ip := range ips {
-		if ip.Is4() {
-			res = append(res, ip)
-		}
-	}
-	return res
-}
-
-func workaroundRequiredIPv6Address(ip netip.Addr) netip.Addr {
-	if ip.IsValid() {
-		return ip
-	}
-
-	var data [16]byte
-	_, err := crand.Read(data[:])
-	utilruntime.Must(err)
-
-	return netip.AddrFrom16(data)
-}
-
 func getIP(ipFamily corev1.IPFamily, ipFamilies []corev1.IPFamily, ips []metalnetv1alpha1.IP) netip.Addr {
 	for i, family := range ipFamilies {
```

Comment:
since right now the `IPs` in the interface CRD is allowed to contain two items/IPs, it does not seem there is a mechanism to prevent/warn two IPv4 or IPv6 addresses, which could due to misconfiguration. I assume it is difficult to achieve such check via the validation mechanism, but using a function to check this case should be feasible.

Reply:
Let me prepare some validation logic to make this more robust.